clients/src/test/java/org/apache/kafka/common/metrics/OtlpMetricsReporter.java

```diff
@@ -0,0 +1,190 @@
+        if (clientLabelsConfig != null) {
+            appendClientLabels = (boolean) clientLabelsConfig;
+        }
+
+        grpcService.initialize();
+    }
+
+
+    @Override
+    public void contextChange(MetricsContext metricsContext) {
+        this.metricsContext = metricsContext.contextLabels();
+    }
+
+
+    @Override
+    public void reconfigure(Map<String, ?> configs) {
+        MetricsReporter.super.reconfigure(configs);
+    }
+
+    @Override
+    public void close() {
+        grpcService.close();
+    }
+
+    @Override
+    public ClientTelemetryReceiver clientReceiver() {
+        return (context, payload) -> {
+
+            if (payload == null || payload.data() == null) {
+                log.warn("exportMetrics - Client did not include payload when pushing to broker, suspended export");
+                return;
+            }
+
+            MetricsData metricsData = ClientTelemetryUtils.deserializeMetricsData(payload.data());
+            if (metricsData == null || metricsData.getResourceMetricsCount() == 0) {
+                log.warn("exportMetrics - No metrics available, suspended export");
+                return;
+            }
+
+            List<ResourceMetrics> metrics = metricsData.getResourceMetricsList();
+
+            // enhance metrics with Broker-added labels
+            if (appendClientLabels) {
+                Map<String, String> labels = getClientLabels(context, payload);
+                metrics = enhanceMetrics(metrics, labels);
+            }
+
+            grpcService.export(metrics);
+        };
+    }
+
+
+    /**
+     * Returns a list of ResourceMetrics where each metrics is enhanced by adding client information as
+     * additional resource level attributes
+     * @param resourceMetrics resource metrics sent by client
+     * @param labels broker added labels containing client information
+     * @return enhanced ResourceMetrics list
+     */
+    public List<ResourceMetrics> enhanceMetrics(List<ResourceMetrics> resourceMetrics,
+                                                Map<String, String> labels) {
+
+        List<ResourceMetrics> updatedResourceMetrics = new ArrayList<>();
+
+        resourceMetrics.forEach(rm -> {
+            Resource.Builder resource = rm.getResource().toBuilder();
+
+            labels.forEach((k, v) -> resource.addAttributes(
+                    KeyValue.newBuilder()
+                            .setKey(k)
+                            .setValue(AnyValue.newBuilder().setStringValue(v))
+                            .build()
+                    ));
+
+            ResourceMetrics updatedMetric = rm.toBuilder()
+                    .setResource(resource.build())
+                    .build();
+
+            updatedResourceMetrics.add(updatedMetric);
+        });
+
+        return updatedResourceMetrics;
+    }
+
+    public static class ClientLabels {
+        public static final String RESOURCE_LABEL_PREFIX = "";
```

Comment:
This blank `RESOURCE_LABEL_PREFIX` doesn't appear to add anything. Is there something it's doing that I don't see?

Reply:
true I'll remove it